x-pack/plugins/cases/server/client/cases/delete.ts

```diff
@@ -49,49 +51,25 @@ export async function deleteCases(ids: string[], clientArgs: CasesClientArgs): P
       entities: Array.from(entities.values()),
     });
 
-    const deleteCasesMapper = async (id: string) =>
-      caseService.deleteCase({
-        id,
-        refresh: false,
-      });
-
-    // Ensuring we don't too many concurrent deletions running.
-    await pMap(ids, deleteCasesMapper, {
-      concurrency: MAX_CONCURRENT_SEARCHES,
+    const attachmentIds = await attachmentService.getAttachmentIdsForCases({
+      caseIds: ids,
+      unsecuredSavedObjectsClient,
     });
 
-    const getCommentsMapper = async (id: string) =>
-      caseService.getAllCaseComments({
-        id,
-      });
+    const userActionIds = await userActionService.getUserActionIdsForCases(ids);
 
-    // Ensuring we don't too many concurrent get running.
-    const comments = await pMap(ids, getCommentsMapper, {
-      concurrency: MAX_CONCURRENT_SEARCHES,
-    });
+    const bulkDeleteEntities: SavedObjectsBulkDeleteObject[] = [
+      ...ids.map((id) => ({ id, type: CASE_SAVED_OBJECT })),
+      ...attachmentIds.map((id) => ({ id, type: CASE_COMMENT_SAVED_OBJECT })),
+      ...userActionIds.map((id) => ({ id, type: CASE_USER_ACTION_SAVED_OBJECT })),
+    ];
 
-    /**
-     * This is a nested pMap.Mapper.
-     * Each element of the comments array contains all comments of a particular case.
-     * For that reason we need first to create a map that iterate over all cases
-     * and return a pMap that deletes the comments for that case
-     */
-    const deleteCommentsMapper = async (commentRes: SavedObjectsFindResponse<CommentAttributes>) =>
-      pMap(commentRes.saved_objects, (comment) =>
-        attachmentService.delete({
-          unsecuredSavedObjectsClient,
-          attachmentId: comment.id,
-          refresh: false,
-        })
-      );
-
-    // Ensuring we don't too many concurrent deletions running.
-    await pMap(comments, deleteCommentsMapper, {
-      concurrency: MAX_CONCURRENT_SEARCHES,
+    await caseService.bulkDeleteCaseEntities({
+      entities: bulkDeleteEntities,
+      options: { refresh: 'wait_for' },
```

Comment:
Since no user action is being created after this we need to `wait_for` the refresh to occur.